target/linux/ipq806x/base-files/etc/board.d/02_network

```diff
@@ -98,6 +98,12 @@ ipq806x_setup_macs()
 			hw_mac_addr=$(mtd_get_mac_ascii hwconfig HW.WAN.MAC.Address)
 			ucidef_set_interface_macaddr "wan" "$hw_mac_addr"
 		;;
+		compex,wpq864)
+			hw_mac_addr=$(mtd_get_mac_ascii 0:APPSBLENV ethaddr)
```

Comment:
You can get MAC-s from U-Boot env via NVMEM instead

Reply:
Thanks for the hint.
It works.  :-)
Is this right?

```
diff --git a/target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts b/target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts
index f28bb92539..a552891946 100644
--- a/target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts
+++ b/target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts
@@ -183,8 +183,11 @@
                        };
 
                        partition@1180000 {
+                               compatible = "u-boot,env";
                                label = "0:APPSBLENV";
                                reg = <0x1180000 0x0080000>;
+                               macaddr_ubootenv_ethaddr: ethaddr {};
+                               macaddr_ubootenv_eth1addr: eth1addr {};
                        };
 
                        partition@1200000 {
@@ -380,6 +383,8 @@
 
        phy-mode = "rgmii";
        qcom,id = <1>;
+       nvmem-cells = <&macaddr_ubootenv_ethaddr>;
+       nvmem-cell-names = "mac-address";
 
        fixed-link {
                speed = <1000>;
@@ -392,6 +397,8 @@
 
        phy-mode = "sgmii";
        qcom,id = <2>;
+       nvmem-cells = <&macaddr_ubootenv_eth1addr>;
+       nvmem-cell-names = "mac-address";
 
        fixed-link {
                speed = <1000>;
```

Reply:
Look OK to me, if it works I would really prefer it instead of userspace